### DISCUSSION TITLE
How to install Ubuntu MATE 22.04 on an Intel mid-2011 iMac

### DISCUSSION THREAD

**Discussion by aki-k** [utmapp/UTM#5218] [Linux] [answered]
I wasn't able to boot the install of Ubuntu MATE 22.04 with UTM using qemu emulation. I tested different graphics options. It always complained about missing metal support. This Intel mid-2011 iMac is updated to Monterey 12.6.5 with help of OpenCore Legacy Patcher. The graphics card is AMD Radeon HD 6770M.

Here's the recipe that enabled me to install Ubuntu MATE 22.04 with qemu installed from homebrew:
```
mkdir -p "${HOME}/.qemu/Ubuntu MATE"
dd if=/dev/zero of="${HOME}/.qemu/Ubuntu MATE/Ubuntu MATE.img" bs=1024k count=32768

qemu-system-x86_64 \
-display cocoa \
-machine type=pc-q35-7.2,vmport=off,i8042=off \
-accel hvf \
-cpu Skylake-Client \
-smp 4 \
-m 4096 \
-device virtio-net,netdev=vmnic \
-netdev user,id=vmnic,hostfwd=tcp::5222-:22 \
-vga virtio \
-audiodev coreaudio,id=coreaudio \
-device ich9-intel-hda -device hda-output,audiodev=coreaudio \
-usb -device usb-tablet,bus=usb-bus.0 \
-device usb-mouse,bus=usb-bus.0 \
-device usb-kbd,bus=usb-bus.0 -device nec-usb-xhci,id=usb-controller-0 \
-drive format=raw,file="${HOME}/.qemu/Ubuntu MATE/Ubuntu MATE.img" \
-drive file="${HOME}/Downloads/ubuntu-mate-22.04.2-desktop-amd64.iso",media=cdrom,readonly=on \
-boot menu=on

press e in grub menu
remove kernel parameters: quiet splash
add kernel parameter: video=1920x1080

(missing xorg pointer):

once in the live session, open a terminal, sudo -i

add file /etc/X11/xorg.conf.d/20-modeset.conf:

Section "Device"
    Identifier "modeset"
    Option     "SWCursor" "on"
EndSection

systemctl restart lightdm

start the install
```

**Comment by osy** [comment, marked as answer]
Why do you want emulation? Virtualization is faster and is what you used in QEMU. Use the wizard, select virtualization, choose your ISO and install it.

**Reply by aki-k** [reply]
If I booted the iso image with virtualization, xorg failed to start at all. The iso booted into Linux virtual console.

Edit: I probably remembered this wrong, and only get a black blank terminal.

**Reply by osy** [reply]
Are you on the latest UTM? Make sure your graphics card doesn't have -gl and you did not select experimental GPU acceleration.

**Reply by aki-k** [reply]
Yes I just downloaded the UTM dmg today. I'll test it again later today.

**Reply by aki-k** [reply]
> Make sure your graphics card doesn't have -gl and you did not select experimental GPU acceleration.

I can't see these settings if I select Apple Virtualization when running the create VM wizard and afterwards.

**Comment by aki-k** [comment]
Now that I configured the Ubuntu MATE 22.04 VM for Apple Virtualization and started it, I only get a black "terminal" with a blinking cursor. Because there's a requirement to give the config a kernel image, I copied both vmlinuz and initrd from the same Ubuntu MATE 22.04 ISO image.

**Comment by aki-k** [comment]
macOS just decided to crap on me when creating two new partitions in Disk Utility so now I can't boot it any more.
